packages/util-endpoints/src/utils/evaluateCondition.ts

```diff
@@ -0,0 +1,13 @@
+import { ConditionObject, EndpointError, EvaluateOptions } from "../types";
+import { callFunction } from "./callFunction";
+
+export const evaluateCondition = ({ assign, ...fnArgs }: ConditionObject, options: EvaluateOptions) => {
+  if (assign && options.referenceRecord[assign]) {
```

Comment:
I suppose this value is never falsy when present, but this could be clearer as `assign in options.referenceRecord`

Reply:
Changed in follow-up PR [`da6c42f` (#3908)](https://github.com/aws/aws-sdk-js-v3/pull/3908/commits/da6c42f7d7f7e88f6dcba6455e008945ef99ddc1)